src/Streamly/System/Process.hs

```diff
@@ -249,3 +326,49 @@ transformChunks ::
     -> t m (Array Word8)    -- ^ Output Stream
 transformChunks fpath args inStream = 
     withInpExe fpath args (AS.concat inStream) FH.toChunks
+
+-- |
+-- Runs a process specified by the path to executable, arguments
+-- that are to be passed and input to be provided to the process
+-- (standard input) in the form of a stream of Word8 and folds
+-- the error stream using the given Fold along with returning
+-- the output of the process (standard output) in the form of a 
+-- stream of Word8
+--
+-- Raises an exception 'ProcessFailed' if process failed due to some
+-- reason. The Fold would continue if you would catch the thrown exception.
+--
+-- @since 0.1.0.0
+{-# INLINE thruExe #-}
+thruExe :: 
+    (IsStream t, MonadIO m, MonadCatch m, MonadAsync m)
+    => FilePath         -- ^ Path to executable
+    -> [String]         -- ^ Arguments to pass to executable
+    -> Fold m Word8 b   -- ^ Fold to fold Error Stream
+    -> t m Word8        -- ^ Input Stream
+    -> t m Word8        -- ^ Output Stream
+thruExe fpath args fld inStream =
+    withErrExe fpath args fld inStream FH.toBytes
+
+-- |
+-- Runs a process specified by the path to executable, arguments
+-- that are to be passed and input to be provided to the process
+-- (standard input) in the form of a array of Word8 and folds
+-- the error stream using the given Fold along with returning
+-- the output of the process (standard output) in the form of a 
+-- array of Word8
+--
+-- Raises an exception 'ProcessFailed' if process failed due to some
+-- reason. The Fold would continue if you would catch the thrown exception.
+--
+-- @since 0.1.0.0
+{-# INLINE thruExeChunks #-}
+thruExeChunks :: 
+    (IsStream t, MonadIO m, MonadCatch m, MonadAsync m)
+    => FilePath                 -- ^ Path to executable
+    -> [String]                 -- ^ Arguments to pass to executable
+    -> Fold m Word8 b           -- ^ Fold to fold Error Stream
+    -> t m (Array Word8)        -- ^ Input Stream
+    -> t m (Array Word8)        -- ^ Output Stream
+thruExeChunks fpath args fld inStream =
```

Comment:
rename this to `transformChunks` and the existing `transformChunks` to `transformChunks_`.